platform/macos/detect.py

```diff
@@ -54,11 +56,13 @@ def ver_parse(a):
         return [int_or_zero(i) for i in a.split(".")]
 
     dirname = os.path.expanduser("~/VulkanSDK")
-    files = os.listdir(dirname)
+    if not os.path.exists(dirname):
+        return None
```

Comment:
I am not sure which versions of python we / SCons expected to support, `typing` is 3.5+ features, if we still have to support older versions it might be better to `return ""` instead (also consistent with other functions).

Reply:
We support 3.5 in the `3.x` branch and 3.6+ in `master`.

That being said, the current conventions would indeed be more to return an empty string and check that - in far most cases it's not a valid value so we're used to use that over things like `Optional`.

I think using `Optional` could be agreed upon but should be part of a codebase-wide refactoring that probably requires a proposal and discussion.

Reply:
Since this function works with paths, I'd be slightly worried to consider empty string as an "error" value, since empty string might be a legit value in some cases (eg. empty path prefix?). So `None` feels like a better value for an error here.